burn-train/src/learner/early_stopping.rs

```diff
@@ -0,0 +1,185 @@
+use crate::metric::{
+    store::{Aggregate, Direction, EventStoreClient, Split},
+    Metric,
+};
+
+/// The condition that [early stopping strategies](EarlyStoppingStrategy) should follow.
+pub enum StoppingCondition {
+    /// When no improvement has happened since the given number of epochs.
+    /// In other words, when no best epoch has been found.
+    NoImprovementSince {
+        /// The number of epochs allowed to get worsen before it get better.
+        n_epochs: usize,
+    },
+}
+
+/// A strategy that checks if the training should be stopped.
+pub trait EarlyStoppingStrategy {
+    /// Update its current state and returns if the training should be stopped.
+    fn should_stop(&mut self, epoch: usize, store: &EventStoreClient) -> bool;
+}
+
+/// An [early stopping strategy](EarlyStoppingStrategy) based on a metrics collected
+/// during training or validation.
+pub struct MetricEarlyStoppingStrategy {
+    condition: StoppingCondition,
+    metric_name: String,
+    aggregate: Aggregate,
+    direction: Direction,
+    split: Split,
+    best_epoch: usize,
+    best_value: f64,
+}
+
+impl EarlyStoppingStrategy for MetricEarlyStoppingStrategy {
+    fn should_stop(&mut self, epoch: usize, store: &EventStoreClient) -> bool {
+        let current_value =
+            match store.find_metric(&self.metric_name, epoch, self.aggregate, self.split) {
+                Some(value) => value,
+                None => {
+                    log::warn!("Can't find metric for early stopping.");
+                    return false;
+                }
+            };
+
+        let is_best = match self.direction {
+            Direction::Lowest => current_value <= self.best_value,
```

Comment:
Shouldn't those be _strictly_ less than / greater than?
Because now even if there is no improvement and stays the same, it will be considered as improvement